Lib/ufo2ft/featureWriters/gdefFeatureWriter.py

```diff
@@ -0,0 +1,92 @@
+from ufo2ft.featureWriters import BaseFeatureWriter, ast
+
+
+class GdefFeatureWriter(BaseFeatureWriter):
+    """Generates a GDEF table based on OpenType Category and glyph anchors.
+
+    It skips generating the GDEF if a GDEF is defined in the features.
+
+    It uses the 'public.openTypeCategories' values to create the GDEF ClassDefs
+    and the ligature caret anchors to create the GDEF ligature carets.
+
+    """
+
+    tableTag = "GDEF"
+
+    def setContext(self, font, feaFile, compiler=None):
```

Comment:
Is it possible to skip all the setup in this method if `if ast.findTable(self.context.feaFile, "GDEF") is not None`? I find it unfortunate to do all the work up front and then just discard it...

Reply:
It's done first and will skip collecting anchors and categories if there now.